request/User/UserHistoryRequest.php

```diff
@@ -36,7 +36,7 @@ public function __construct(string $username, $type = '') // "" means empty stri
         $this->username = $username;
 
         if (null !== $type) {
-            if (!\in_array($type, ['anime', 'manga'])) {
+            if (!empty($type) && !\in_array($type, ['anime', 'manga'])) {
```

Comment:
please don't use empty, but check the specific value.

Reply:
> please don't use empty, but check the specific value.

Would you prefer an explicit check along with the other supported types?

```suggestion
            if (!\in_array($type, ['', 'anime', 'manga'])) {
```